Python/pylifecycle.c

```diff
@@ -1777,6 +1811,8 @@ Py_FinalizeEx(void)
     _PyAtExit_Call(tstate->interp);
     PyUnstable_PerfMapState_Fini();
 
+    wait_for_exit_blocks_to_be_released();
```

Comment:
This should apply to each interpreter (see `Py_EndInterpreter()`), not just the main interpreter at global runtime finalization, no?